ktor-features/ktor-jwt/src/org/jetbrains/ktor/jwt/JWTAuth.kt

```diff
@@ -0,0 +1,96 @@
+package org.jetbrains.ktor.jwt
+
+import com.auth0.jwk.Jwk
+import com.auth0.jwk.JwkProvider
+import com.auth0.jwt.JWT
+import com.auth0.jwt.JWTVerifier
+import com.auth0.jwt.algorithms.Algorithm
+import com.auth0.jwt.impl.JWTParser
+import com.auth0.jwt.interfaces.DecodedJWT
+import com.auth0.jwt.interfaces.Payload
+import org.jetbrains.ktor.application.ApplicationCall
+import org.jetbrains.ktor.auth.AuthenticationPipeline
+import org.jetbrains.ktor.auth.Credential
+import org.jetbrains.ktor.auth.NotAuthenticatedCause
+import org.jetbrains.ktor.auth.Principal
+import org.jetbrains.ktor.http.HttpHeaders
+import org.jetbrains.ktor.http.HttpStatusCode
+import org.jetbrains.ktor.response.respond
+import java.security.interfaces.ECPublicKey
+import java.security.interfaces.RSAPublicKey
+import java.util.*
+
+val JWTAuthKey: Any = "JWTAuth"
+
+data class JWTCredential(val payload: Payload): Credential
+data class JWTPrincipal(val payload: Payload): Principal
+
+fun AuthenticationPipeline.jwtAuthentication(verifier: JWTVerifier, validate: (JWTCredential) -> Principal?) {
+    intercept(AuthenticationPipeline.RequestAuthentication) { context ->
+        try {
+            val token = call.getAuthToken()
+
+            val jwt = verifier.verify(token)
+            val payload = jwt.parsePayload()
+
+            val credentials = org.jetbrains.ktor.jwt.JWTCredential(payload)
+            val principal = requireNotNull(credentials.let(validate))
+            context.principal(principal)
+        } catch(e: Exception) {
```

Comment:
Catching general exception is not really good. What kinds of exceptions are expected here?